pkg/operator/encryption/statemachine/transition.go

```diff
@@ -171,7 +171,26 @@ func getDesiredEncryptionState(oldEncryptionConfig *apiserverconfigv1.Encryption
 	}
 
 	//
-	// STEP 4: with consistent read-keys and write-keys, remove every read-key other than the write-key and one last read key.
+	// STEP 4: with consistent keys remove a GR that this operator no longer manages
+	//
+	// TODO: what if len(toBeEncryptedGRs)  == 0
```

Comment:
do we need an additional check?

Reply:
I don't think we need that. In that case we remove the whole encryption mechanism.